### ISSUE TITLE
router executor crash (assert) on non-colocated joins between range tables

### ISSUE THREAD

**Issue by mtuncer** [citusdata/citus#2057]
Hit this when investigating https://github.com/citusdata/citus/issues/1284 

I had 4 worker nodes, therefore when you create a 2 shards per table they won't be co-located.

```sql
CREATE TABLE events (
	user_id int,
	event_id bigint,
	event_type character varying(255),
	event_time bigint
);
SELECT master_create_distributed_table('events', 'user_id', 'range');

SELECT master_create_empty_shard('events') AS new_shard_id
\gset
UPDATE pg_dist_shard SET shardminvalue = 1, shardmaxvalue = 100
WHERE shardid = :new_shard_id;

SELECT master_create_empty_shard('events') AS new_shard_id
\gset
UPDATE pg_dist_shard SET shardminvalue = 101, shardmaxvalue = 200
WHERE shardid = :new_shard_id;

CREATE TABLE users (
	user_id int,
	lastseen bigint
);

SELECT master_create_distributed_table('users', 'user_id', 'range');

SELECT master_create_empty_shard('users') AS new_shard_id
\gset
UPDATE pg_dist_shard SET shardminvalue = 1, shardmaxvalue = 100
WHERE shardid = :new_shard_id;

SELECT master_create_empty_shard('users') AS new_shard_id
\gset
UPDATE pg_dist_shard SET shardminvalue = 101, shardmaxvalue = 200
WHERE shardid = :new_shard_id;
```
following query would be a router query on co-located tables
```sql
select * from users, events where users.user_id = events.user_id and users.user_id = 5;
```

Since tables are not co-located it can't be router planned. Following output is get instead (client_min_messages = DEBUG2)

```
DEBUG:  Found no worker with all shard placements
DEBUG:  join prunable for intervals [1,100] and [101,200]
DEBUG:  Plan is router executable
server closed the connection unexpectedly
	This probably means the server terminated abnormally
	before or while processing the request.
The connection to the server was lost. Attempting reset: Failed
```

**Comment by mtuncer** [comment]
This seems to be fixed in 7.4

**Comment by onderkalaci** [comment]
> This seems to be fixed in 7.4

It works fine on master (as of 9.3) so closing this. (and we don't have router executor anymore)
